Backpack/FloatingNotification/Classes/FloatingNotificationViewModel.swift

```diff
@@ -0,0 +1,29 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+struct FloatingNotificationViewModel {
```

Comment:
could we make this one public and have clients send an instance of this directly?

in the future (or now) we could have static functions that only receive certain params and give it specific names, like:
```swift
public struct FloatingNotificationViewModel {
   let ... 
   // all the properties..
   public init(...) {...}

   public static func titleOnly(_ title: String) -> FloatingNotificationViewModel {}
   public static func titleWithAction(_ title: String, action: FloatingNotificationAction) -> FloatingNotificationViewModel {}
}

// So when showing a notification, it could look like this from the client

notification.show(.titleOnly("Howdy!"))
```

Reply:
Yeah sounds good. Had the same init as Android to align but happy to change it to this